### ISSUE TITLE
Add triplet for training models

### ISSUE THREAD

**Issue by lutianhao** [open-mmlab/mmpose#1529]
Hi，If I want to add triplet for training how should I add it to the project in a reasonable way?

**Comment by jin-s13** [comment]
could you please describe **triplet** in more details?

**Comment by lutianhao** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        I'd like to add triplet loss as a train loss, It is hoped that images with different angles of the same ID can learn similar features. 
    
        The source code I plan to use for Tripletloss is : torch.nn — PyTorch master documentation
    
     
    
        
                    ***@***.***
        
    
 

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        8/1/2022 18:16
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [open-mmlab/mmpose] Add triplet for training models (Issue #1529)
        
      
    
  
  
could you please describe triplet in more details?

**Comment by jin-s13** [comment]
Maybe you can add a custom dataset to generate triplets. I am not sure.

**Comment by lutianhao** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        But how to deal with batch size. for I've read 'bottomup coco dataset' and it just catch data by function 'get_single', how mmpose load new data? like use forward or iter function? Could you please point the specific code files and line numbers? Thanks!
    
        
            
        
    
    
     
    
        
                    ***@***.***
        
    
 

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        8/1/2022 18:32
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [open-mmlab/mmpose] Add triplet for training models (Issue #1529)
        
      
    
  
  
Maybe you can add a custom dataset to generate triplets. I am not sure.
